### PR TITLE
Added support for Polarberry

### PR DESCRIPTION
**Pull request by Mans-sdsp**
Support for Polarberry added to the 2023.07.next branch

### CODE REVIEW

File: arch/riscv/dts/sundancedsp_polarberry.dts [hunk 1]
```diff
@@ -0,0 +1,229 @@
+/dts-v1/;
+
+#include "microchip-mpfs.dtsi"
+
+/* Clock frequency (in Hz) of the rtcclk */
+#define RTCCLK_FREQ		1000000
+
+/ {
+	model = "Microchip POLARBERRY";
```
Comment (ConchuOD) [inline arch/riscv/dts/sundancedsp_polarberry.dts]: It's a sundance polarberry, not a microchip polarberry, right?

File: arch/riscv/dts/sundancedsp_polarberry.dts [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+	compatible = "microchip,mpfs";
```
Comment (ConchuOD) [inline arch/riscv/dts/sundancedsp_polarberry.dts]: You need to add a board-specific compatible here. You can check the linux repo & use the one listed there for the polarberry.

File: arch/riscv/dts/sundancedsp_polarberry.dts [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+	disable-wp;
+};
+
+&i2c0 {
+	uboot,dm-pre-reloc;
+
+	status = "okay";
+	clock-frequency = <100000>;
+
+	// 0x72 = SI538A-B-GM
```
Comment (ConchuOD) [inline arch/riscv/dts/sundancedsp_polarberry.dts]: What does this comment refer to?

File: arch/riscv/dts/sundancedsp_polarberry.dts [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+&uart0 {
+	status = "okay";
+};
+
+&uart1 {
+	status = "okay";
+};
+
+&mmc {
+	uboot,dm-pre-reloc;
```
Comment (ConchuOD) [inline arch/riscv/dts/sundancedsp_polarberry.dts]: These u-boot specific bits look like they should be in sundancedsp_polarberry-u-boot.dtsi

File: arch/riscv/dts/sundancedsp_polarberry.dts [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+
+	status = "okay";
+	clock-frequency = <100000>;
+};
+
+&mac1 {
+	status = "okay";
+
+	local-mac-address = [56 34 00 FC 00 03];
```
Comment (ConchuOD) [inline arch/riscv/dts/sundancedsp_polarberry.dts]: Why is a fixed mac address being set in the dts? It looks like you have included the code in your platform that will create a mac address at runtime depending on board serial number?

File: arch/riscv/dts/sundancedsp_polarberry.dts [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+	};
+};
+
+&usb {
+	uboot,dm-pre-reloc;
+
+	status = "okay";
+	xlnx,tz-nonsecure = <0x1>;
+	xlnx,usb-polarity = <0x0>;
+	xlnx,usb-reset-mode = <0x2>;
```
Comment (ConchuOD) [inline arch/riscv/dts/sundancedsp_polarberry.dts]: What affect do these xlnx vendor properties have on our usb controller?

File: arch/riscv/dts/sundancedsp_polarberry.dts [hunk 7]
```diff
@@ -0,0 +1,229 @@
+/dts-v1/;
```
Comment (ConchuOD) [inline arch/riscv/dts/sundancedsp_polarberry.dts]: re: the filename, can you call it "mpfs-polarberry.dts" please? I know the icicle file has the vendor in the name, but we're moving away from that at the moment, and have moved most of the boards to having no vendor in the name. In upstream u-boot, the ones prefixed by "microchip" have had that removed, and next time we bump the version of our vendor u-boot the vendor prefixes will vanish.

**Review by ConchuOD — CHANGES_REQUESTED** [review]
Hey Mans,

Apologies for the delay, been busy with release activities unfortunately.
I've left some comments, mainly on the devicetree, but overall this looks pretty good to me.

Thanks,
Conor.

File: arch/riscv/dts/sundancedsp_polarberry.dts (hunk 2 above)
Comment (Mans-sdsp) [reply arch/riscv/dts/sundancedsp_polarberry.dts]: Hello Conor, 
Can you please elaborate? Polarberry is compatible with Microchip, mpfs, what do we put here?
Kisn regards

Comment (ConchuOD) [reply arch/riscv/dts/sundancedsp_polarberry.dts]: https://github.com/linux4microchip/linux/blob/linux-6.1-mchp/arch/riscv/boot/dts/microchip/mpfs-polarberry.dts#L14

**Comment by Mans-sdsp** [comment]
Hello Conor,
   The requested changes were implemented and tested. Please check the PR and merge it if there are no other comments so we can start working on Buildroot PR.

File: arch/riscv/dts/mpfs-polarberry.dts [hunk 8]
```diff
@@ -0,0 +1,219 @@
+/dts-v1/;
+
+#include "microchip-mpfs.dtsi"
+
+/* Clock frequency (in Hz) of the rtcclk */
+#define RTCCLK_FREQ		1000000
+
+/ {
+	model = "SundanceDSP Polarberry";
+	compatible = "microchip,mpfs";
```
Comment (ConchuOD) [inline arch/riscv/dts/mpfs-polarberry.dts]: Please use         `compatible = "sundance,polarberry", "microchip,mpfs";` like the linux devicetree

File: arch/riscv/dts/mpfs-polarberry.dts [hunk 9]
```diff
@@ -0,0 +1,219 @@
...
+	clock-frequency = <100000>;
+};
+
+&mac1 {
+	status = "okay";
+
+	/*
+	    You can forcibly set MAC addres in "local-mac-address" (maybe usefull for debug)
+	*/
+//	local-mac-address = [56 34 00 FC 00 03];
```
Comment (ConchuOD) [inline arch/riscv/dts/mpfs-polarberry.dts]: Please remove these comments, the DT is not a suitable place for explaining how to use properties. You've included the code that will populate the mac address as an environment variable, based on the serial number of the SoC.
Either the contents of the environment variable, or a user-controlled value, can be set as the mac address in the u-boot console.

**Review by ConchuOD — CHANGES_REQUESTED** [review]
I left a couple of comments on the dts, but could you also rebase/squash this PR so that it is a single commit?
I'd be happy to merge this once those are resolved.

**Comment by Mans-sdsp** [comment]
Hello Conor, I made all the required changes. Please check it.

**Comment by Mans-sdsp** [comment]
it seems it is not merged yet!

**Comment by Mans-sdsp** [comment]
@ConchuOD It doesn't seem to be merged.

**Comment by ConchuOD** [comment]
> @ConchuOD It doesn't seem to be merged.

I'm not sure that I understand - you closed and reopened this PR several times over the last few days, but none of them contained any commits with polarberry content as far as I can tell. Please check that you pushed the right branch back to github after you did your rebase.

Cheers,
Conor.

**Comment by Mans-sdsp** [comment]
It is very strange! Please see the changes.

**Comment by ConchuOD** [comment]
> It is very strange! Please see the changes.

That looks a lot better, thanks!

**Pull request merged**